x/group/keeper/keeper_test.go

```diff
@@ -2416,22 +2422,206 @@ func (s *TestSuite) TestExecProposal() {
 			}
 			s.Require().NoError(err)
 
-			// and proposal is updated
-			res, err := s.keeper.Proposal(ctx, &group.QueryProposalRequest{ProposalId: proposalID})
+			if !(spec.expExecutorResult == group.PROPOSAL_EXECUTOR_RESULT_SUCCESS) {
+
+				// and proposal is updated
+				res, err := s.keeper.Proposal(ctx, &group.QueryProposalRequest{ProposalId: proposalID})
+				s.Require().NoError(err)
+				proposal := res.Proposal
+
+				exp := group.ProposalResult_name[int32(spec.expProposalResult)]
+				got := group.ProposalResult_name[int32(proposal.Result)]
+				s.Assert().Equal(exp, got)
+
+				exp = group.ProposalStatus_name[int32(spec.expProposalStatus)]
+				got = group.ProposalStatus_name[int32(proposal.Status)]
+				s.Assert().Equal(exp, got)
+
+				exp = group.ProposalExecutorResult_name[int32(spec.expExecutorResult)]
+				got = group.ProposalExecutorResult_name[int32(proposal.ExecutorResult)]
+				s.Assert().Equal(exp, got)
+			}
+
+			if spec.expBalance {
+				fromBalances := s.app.BankKeeper.GetAllBalances(sdkCtx, s.groupPolicyAddr)
+				s.Require().Contains(fromBalances, spec.expFromBalances)
+				toBalances := s.app.BankKeeper.GetAllBalances(sdkCtx, addr2)
+				s.Require().Contains(toBalances, spec.expToBalances)
+			}
+		})
+	}
+}
+
+func (s *TestSuite) TestExecPrunedProposals() {
+	addrs := s.addrs
+	addr1 := addrs[0]
+	addr2 := addrs[1]
+
+	msgSend1 := &banktypes.MsgSend{
+		FromAddress: s.groupPolicyAddr.String(),
+		ToAddress:   addr2.String(),
+		Amount:      sdk.Coins{sdk.NewInt64Coin("test", 100)},
+	}
+	msgSend2 := &banktypes.MsgSend{
+		FromAddress: s.groupPolicyAddr.String(),
+		ToAddress:   addr2.String(),
+		Amount:      sdk.Coins{sdk.NewInt64Coin("test", 10001)},
+	}
+	proposers := []string{addr2.String()}
+	specs := map[string]struct {
+		srcBlockTime      time.Time
+		setupProposal     func(ctx context.Context) uint64
+		expErr            bool
+		expErrMsg         string
+		expProposalStatus group.ProposalStatus
+		expProposalResult group.ProposalResult
+		expExecutorResult group.ProposalExecutorResult
+		expBalance        bool
+		expFromBalances   sdk.Coin
+		expToBalances     sdk.Coin
```

Comment:
Let's keep this test focused on pruning after Exec. We don't need these 6 fields for this test, we just need to check:
- if proposal passes, then propasal and votes are pruned
- else, proposal and votes are still accessible